components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/bindings/impl/SSOSessionBasedTokenBinder.java

```diff
@@ -156,4 +150,56 @@ public boolean isValidTokenBinding(OAuth2AccessTokenReqDTO oAuth2AccessTokenReqD
 
         return isValidTokenBinding(oAuth2AccessTokenReqDTO, bindingReference, COMMONAUTH_COOKIE);
     }
+
+    @Override
+    public boolean isValidTokenBinding(AccessTokenDO accessTokenDO) {
+
+        if (accessTokenDO == null || accessTokenDO.getTokenBinding() == null ||
+                StringUtils.isBlank(accessTokenDO.getTokenBinding().getBindingValue())) {
+            if (log.isDebugEnabled()) {
+                log.debug("No token binding value is found for SSO session bound token.");
+            }
+            return false;
+        }
+        String sessionIdentifier = accessTokenDO.getTokenBinding().getBindingValue();
+        return isValidSession(sessionIdentifier, resolveTenantDomain(accessTokenDO));
+    }
+
+    /**
+     * Checks if the session is valid by retrieving the session context information using the session identifier.
+     *
+     * @param sessionIdentifier The session identifier from the token binding.
+     * @param tenantDomain      The tenant domain extracted from the access token owner.
+     * @return true if the session is valid, false otherwise.
+     */
+    private boolean isValidSession(String sessionIdentifier, String tenantDomain) {
+
+        SessionContext sessionContext = FrameworkUtils.getSessionContextFromCache(sessionIdentifier, tenantDomain);
+        if (sessionContext == null) {
+            if (log.isDebugEnabled()) {
+                log.debug("Session context is not found corresponding to the session identifier: " +
+                        sessionIdentifier);
+            }
+            return false;
+        }
+
+        if (log.isDebugEnabled()) {
+            log.debug("SSO session validation successful for the given session identifier: " + sessionIdentifier);
+        }
+        return true;
+    }
+
+    /**
+     * Resolve the tenant domain.
+     *
+     * @param accessTokenDO The access token data object.
+     * @return The tenant domain.
+     */
+    private String resolveTenantDomain(AccessTokenDO accessTokenDO) {
+
+        if (accessTokenDO.getAuthzUser() != null) {
+            return accessTokenDO.getAuthzUser().getTenantDomain();
+        }
+        return FrameworkUtils.getLoginTenantDomainFromContext();
```

Comment:
Shall we check this behaviour for cross tenant introspection

Reply:
the fallback ` return FrameworkUtils.getLoginTenantDomainFromContext();` is redundant as the accessTokenDO with a binding type obviously has Authz-user associated with. Hence removed the usage of `FrameworkUtils.getLoginTenantDomainFromContext()`